packages/features/tips/UpgradeTip.tsx

```diff
@@ -0,0 +1,86 @@
+import { useMemo } from "react";
+import type { ReactNode } from "react";
+
+import { classNames } from "@calcom/lib";
+import { useHasTeamPlan } from "@calcom/lib/hooks/useHasTeamPlan";
+import { useLocale } from "@calcom/lib/hooks/useLocale";
+// import isCalcom from "@calcom/lib/isCalcom";
+import { trpc } from "@calcom/trpc/react";
+import { EmptyScreen, Icon } from "@calcom/ui";
+
+import TeamList from "../ee/teams/components/TeamList";
+
+const isCalcom = true;
+export function UpgradeTip({
+  dark,
+  title,
+  description,
+  background,
+  features,
+  buttons,
+  isParentLoading,
+  children,
+}: {
+  dark?: boolean;
+  title: string;
+  description: string;
+  background: string;
+  features: Array<{ icon: JSX.Element; title: string; description: string }>;
+  buttons?: JSX.Element;
+  /**Chldren renders when the user is in a team */
+  children: JSX.Element;
+  isParentLoading?: ReactNode;
+}) {
+  const { data } = trpc.viewer.teams.list.useQuery();
+
+  const invites = useMemo(() => data?.filter((m) => !m.accepted) || [], [data]);
+  const { t } = useLocale();
+  const hasTeamPlan = useHasTeamPlan();
+
+  if (hasTeamPlan) return children;
+
+  if (!isCalcom)
+    return <EmptyScreen Icon={Icon.FiUsers} headline={title} description={description} buttonRaw={buttons} />;
+
+  if (isParentLoading) return <>{isParentLoading}</>;
```

Comment:
Seems not right to have isParentLoading here.
- I see that isParentLoading is the same component that we are rendering everywhere. So why not have that component rendered here directly.
- If there is a possibility of different components being rendered for different use cases, is it possible to achieve the same by putting this component behind that loader